core/node/proof_data_handler/src/tee_request_processor.rs

```diff
@@ -0,0 +1,134 @@
+use std::sync::Arc;
+
+use axum::{extract::Path, Json};
+use zksync_config::configs::ProofDataHandlerConfig;
+use zksync_dal::{tee_proof_generation_dal::TeeType, ConnectionPool, Core, CoreDal, SqlxError};
+use zksync_object_store::ObjectStore;
+use zksync_prover_interface::api::{
+    GenericProofGenerationDataResponse, RegisterTeeAttestationRequest,
+    RegisterTeeAttestationResponse, SubmitProofResponse, SubmitTeeProofRequest,
+    TeeProofGenerationDataRequest,
+};
+use zksync_tee_verifier::TeeVerifierInput;
+use zksync_types::L1BatchNumber;
+
+use crate::errors::RequestProcessorError;
+
+pub type TeeProofGenerationDataResponse = GenericProofGenerationDataResponse<TeeVerifierInput>;
+
+#[derive(Clone)]
+pub(crate) struct TeeRequestProcessor {
+    blob_store: Arc<dyn ObjectStore>,
+    pool: ConnectionPool<Core>,
+    config: ProofDataHandlerConfig,
+}
+
+impl TeeRequestProcessor {
+    pub(crate) fn new(
+        blob_store: Arc<dyn ObjectStore>,
+        pool: ConnectionPool<Core>,
+        config: ProofDataHandlerConfig,
+    ) -> Self {
+        Self {
+            blob_store,
+            pool,
+            config,
+        }
+    }
+
+    pub(crate) async fn get_proof_generation_data(
+        &self,
+        request: Json<TeeProofGenerationDataRequest>,
+    ) -> Result<Json<TeeProofGenerationDataResponse>, RequestProcessorError> {
+        tracing::info!("Received request for proof generation data: {:?}", request);
+
+        let mut connection = self
+            .pool
+            .connection()
+            .await
+            .map_err(|_| RequestProcessorError::Sqlx(SqlxError::PoolClosed))?;
```

Comment:
Will leave it as a separate comment to not mix new and old discussions. I think the right approach here would be to use `DalError` instead of `SqlxError` in `RequestProcessorError` (I believe that `RequestProcessorError` simply predates `DalError` which is why it wasn't integrated). Then you'll be able to use `inner` inside of `into_response` for the purpose of pattern matching.

Reply:
[Fixed](https://github.com/matter-labs/zksync-era/pull/1993/commits/2b296b3d0c397e2ba72cce49eacb866e917e4d0d). ✅